internal/controller/clientpool/clientpool.go

```diff
@@ -48,13 +48,24 @@ func (cp *ClientPool) GetWorkflowServiceClient(hostPort string) (workflowservice
 	return nil, false
 }
 
+func (cp *ClientPool) GetSDKClient(hostPort string) (client.Client, bool) {
+	cp.mux.RLock()
+	defer cp.mux.RUnlock()
+
+	c, ok := cp.clients[hostPort]
```

Comment:
self-hosted clients can have multiple ns per hostport. so index needs to contain ns also